### PR TITLE
Add OMNeT++ MSG language

### PR DESCRIPTION
**Pull request by adamgeorge309**
This adds the OMNeT++ MSG language.

## Description
MSG (extension .msg) is a DSL for the OMNeT++ network simulator. It is used by model frameworks for OMNeT++, such as the INET framework, Simu5G, Veins, etc. We are the team that develops [OMNeT++](https://github.com/omnetpp/omnetpp) and the [INET Framework](https://github.com/inet-framework/) and we have created a TextMate grammar for this language.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.msg+opensim
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/inet-framework/inet
    - Sample license:
      - LGPLv3
  - [x] I have included a syntax highlighting grammar: https://github.com/omnetpp/omnetpp-textmate-msg
      
  - [x] I have added a color
    - Hex value: `#a0e0a0`
    - Rationale: Matches the branding guidelines for the OMNeT++ project.
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Review by lildude — CHANGES_REQUESTED** [review]
This is a very generic extension which [matches way more files](https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.msg) than your limited search which means they'll all be identified as this language.

As such, you'll need to add this extension to the [`generic.yml`](https://github.com/github-linguist/linguist/blob/master/lib/linguist/generic.yml) and add a very precise heuristic that only matches files for this language.

**Comment by adamgeorge309** [comment]
I did the heuristics. Thanks.

**Comment by adamgeorge309** [comment]
Based on the heuristic, I refined the popularity search:

https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.msg+content%3A%2F%5Ecplusplus+%5C%7B%5C%7B%7C%5Enamespace%5B%5Cs%5D*%28%5B%5E.%5Cs%5D*%5C.%29*%5B%5E.%5Cs%5D*%3B%7C%5Estruct+%5C%7B%7C%5Emessage+%5B%5CS%5D*+%28extends%29%3F+%5B%5CS%5D*%5B%5Cs%5D*%5C%7B%7C%5Epacket+%5C%7B%7C%5Eclass+%28extends%29+%5B%5CS%5D*%5B%5Cs%5D*%5C%7B%7C%5Eenum+%5C%7B%7C%5Eimport+%28%5B%5E.%5Cs%5D*%5C.%29*%5B%5E.%5Cs%5D*%3B%2F

This should be a more accurate search for assessing the popularity. Thank you!

**Review by lildude — DISMISSED** [review]

**Comment by lildude** [comment]
Please merge main into your branch so CI can run. I'd normally do it but as your PR comes from an org-owned fork, permission can't be granted to me.

**Review by lildude — CHANGES_REQUESTED** [review]
These are legit failures:

```
  1) Failure:
TestPedantic#test_heuristics_tests_are_exhaustive [test/test_pedantic.rb:62]:
Extension .msg has no test in test_heuristics.rb
```

We need a test for this heuristic.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - main
+      - omnetpp-msg
```
Comment (lildude) [inline .github/workflows/ci.yml:8]: Please don't modify this file. We don't need this.

**Review by lildude — CHANGES_REQUESTED** [review]
Please remove `samples/omnetpp-msg/Ieee80211Frame.msg` as it's too large and unnecessary as there three other files.

**Review by lildude — APPROVED** [review]
Lovely! Thanks for addresses those issues. 🙇

**Comment by adamgeorge309** [comment]
Thanks

**Pull request merged**